src/System.Devices.Gpio/System.Devices.Gpio.Samples/RgbColorSensor.cs

```diff
@@ -0,0 +1,466 @@
+        {
+            byte enable = Read8(REGISTER_ENABLE);
+
+            if (value == -1)
+            {
+                Write8(REGISTER_ENABLE, (byte)(enable & ~ENABLE_AIEN));
+            }
+            else
+            {
+                int index = Array.IndexOf(CYCLES, value);
+
+                if (index < 0)
+                {
+                    throw new ArgumentOutOfRangeException($"The value of parameter {nameof(value)} must be 0, 1, 2, 3, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55 or 60.");
+                }
+
+                Write8(REGISTER_ENABLE, (byte)(enable | ENABLE_AIEN));
+                Write8(REGISTER_APERS, (byte)index);
+            }
+        }
+
+        /// <summary>
+        /// Gets the minimum threshold value of the sensor.
+        /// </summary>
+        public ushort GetMinValue()
+        {
+            ushort result = Read16(REGISTER_AILT);
+            return result;
+        }
+
+        /// <summary>
+        /// Sets the minimum threshold value of the sensor.
+        /// </summary>
+        public void SetMinValue(ushort value)
+        {
+            Write16(REGISTER_AILT, value);
+        }
+
+        /// <summary>
+        /// Gets the maximum threshold value of the sensor.
+        /// </summary>
+        public ushort GetMaxValue()
+        {
+            ushort result = Read16(REGISTER_AIHT);
+            return result;
+        }
+
+        /// <summary>
+        /// Sets the maximum threshold value of the sensor.
+        /// </summary>
+        public void SetMaxValue(ushort value)
+        {
+            Write16(REGISTER_AIHT, value);
+        }
+
+        /// <summary>
+        /// Check if the status bit is set and the chip is ready.
+        /// </summary>
+        private bool IsValid()
+        {
+            byte result = Read8(REGISTER_STATUS);
+            return result > 0;
+        }
+
+        /// <summary>
+        /// Read the raw RGBC color detected by the sensor.
+        /// Reads a raw color of 16-bit red, green, blue, clear component values (0-65535).
+        /// </summary>
+        private void ReadColor()
+        {
+            bool wasActive = IsActive;
+            SetActive(true);
+
+            while (!IsValid())
+            {
+                SleepIntegrationTime();
+            }
+
+            _rawColor.Red = Read16LittleEndian(REGISTER_RDATA);
+            _rawColor.Green = Read16LittleEndian(REGISTER_GDATA);
+            _rawColor.Blue = Read16LittleEndian(REGISTER_BDATA);
+            _rawColor.Clear = Read16LittleEndian(REGISTER_CDATA);
+
+            SetActive(wasActive);
+
+            // Console.WriteLine($"R = {_rawColor.Red}, G = {_rawColor.Green}, B = {_rawColor.Blue}, C = {_rawColor.Clear}");
+        }
+
+        private void SleepIntegrationTime()
+        {
+            int time = 0;
+
+            switch (_integrationTime)
+            {
+                case RgbColorSensorIntegrationTime.Milliseconds2_4:
```

Comment:
why is this case called 2_4 when all the others have a 1:1 map to their time?

Reply:
It means 2.4 milliseconds, so I rounded it up to 3 milliseconds since Sleep method takes an int as parameter.